fsclient/client.go

```diff
@@ -0,0 +1,169 @@
+package fsclient
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/DAv10195/submit_commons/encryption"
+	"github.com/sirupsen/logrus"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	url2 "net/url"
+)
+
+type FileServerClient struct {
+	adr string
+	username string
+	password string
+	logger   *logrus.Entry
+	encryption encryption.Encryption
+}
+
+func NewFileServerClient(adr string, username string, password string, logger *logrus.Entry, encryption encryption.Encryption) *FileServerClient{
+	encryptedPass, err := encryption.Encrypt(password)
+	if err != nil {
+		panic("error in encryption")
+	}
+	return &FileServerClient{
+		adr: adr,
+		username: username,
+		password: encryptedPass,
+		logger:   logger,
+		encryption: encryption,
+	}
+}
+
+func (fsc *FileServerClient) UploadFile (url string, reader io.Reader, filename string) error {
```

Comment:
isn't it more simple to accept an *os.File instead of a reader and then derive the file name from it instead of accepting them separately?

Reply:
Also, why not accept multiple files with ...

Reply:
replaced with  *os.File
regarding multiple files, as we agreed yesterday that we will write the function on the way if needed.